bgls/simulator_test.py

```diff
@@ -175,3 +175,139 @@ def test_run_with_density_matrix_simulator():
     result_density_matrix = sim_density_matrix.run(circuit, repetitions=100)
 
     assert result_density_matrix == result_state_vector
+
+
+def test_run_with_stabilizer_ch_simulator():
+    """Test sampled bitstrings are same when using a state vector ch form
+    simulator and a statevector simulator.
+    """
+    a, b, c = cirq.LineQubit.range(3)
+    circuit = cirq.Circuit(
+        cirq.H(a),
+        cirq.CNOT(a, b),
+        cirq.X.on(c),
+        cirq.measure([a, b, c], key="z"),
+    )
+    sim_state_vector = bgls.Simulator(
+        cirq.StateVectorSimulationState(qubits=(a, b, c), initial_state=0),
+        cirq.protocols.act_on,
+        bgls.utils.cirq_state_vector_bitstring_probability,
+        seed=1,
+    )
+    result_state_vector = sim_state_vector.run(circuit, repetitions=100)
+
+    sim_stabilizer_ch = bgls.Simulator(
+        cirq.StabilizerChFormSimulationState(
+            qubits=(a, b, c), initial_state=0
+        ),
+        cirq.protocols.act_on,
+        bgls.utils.cirq_stabilizer_ch_bitstring_probability,
+        seed=1,
+    )
+    result_stabilizer_ch = sim_stabilizer_ch.run(circuit, repetitions=100)
+
+    assert result_stabilizer_ch == result_state_vector
+
+
+def test_run_with_stabilizer_ch_simulator_near_clifford():
+    """Test sampled bitstrings are same when using a state vector ch form
+    simulator and a statevector simulator. Using apply_near_clifford_gate
+    can work with Clifford+T circuits as well.
+    """
+    a, b, c = cirq.LineQubit.range(3)
+    circuit = cirq.Circuit(
+        cirq.H(a),
+        cirq.CNOT(a, b),
+        cirq.X.on(c),
+        cirq.T(c),
+        cirq.measure([a, b, c], key="z"),
+    )
+    sim_state_vector = bgls.Simulator(
+        cirq.StateVectorSimulationState(qubits=(a, b, c), initial_state=0),
+        cirq.protocols.act_on,
+        bgls.utils.cirq_state_vector_bitstring_probability,
+        seed=1,
+    )
+    result_state_vector = sim_state_vector.run(circuit, repetitions=100)
+
+    sim_stabilizer_ch = bgls.Simulator(
+        cirq.StabilizerChFormSimulationState(
+            qubits=(a, b, c), initial_state=0
+        ),
+        bgls.utils.apply_near_clifford_gate,
+        bgls.utils.cirq_stabilizer_ch_bitstring_probability,
+        seed=1,
+    )
+    result_stabilizer_ch = sim_stabilizer_ch.run(circuit, repetitions=100)
+
+    assert result_stabilizer_ch == result_state_vector
+
+
+def test_remains_clifford():
+    """Creating a large random circuit of clifford gates, the simulator
+    should remain clifford throughout.
+    """
+    a, b, c = cirq.LineQubit.range(3)
+    domain = {cirq.H: 1, cirq.CNOT: 2, cirq.S: 1}
+    clifford_circuit = cirq.testing.random_circuit(
+        [a, b, c], n_moments=100, op_density=0.5, gate_domain=domain
+    )
+    clifford_circuit = clifford_circuit + cirq.measure([a, b, c], key="z")
+    for op in clifford_circuit.all_operations():
+        assert cirq.has_stabilizer_effect(op)
```

Comment:
This is testing the `cirq.testing.random_circuit` and `cirq.has_stabilizer_effect` functions? You don't need to test code you didn't write.

Reply:
see above comment, removed this portion of the function and just testing act_on_near_clifford instead